galite-core/src/main/kotlin/org/kopi/galite/ui/vaadin/notif/VConfirmNotification.kt

```diff
@@ -17,85 +17,66 @@
 import com.vaadin.flow.component.ShortcutEvent
+import com.vaadin.flow.component.Shortcuts
 import com.vaadin.flow.component.button.Button
-import com.vaadin.flow.component.html.H3
-import com.vaadin.flow.component.html.Span
-import com.vaadin.flow.component.orderedlayout.FlexComponent
-
-import org.kopi.galite.ui.vaadin.base.LocalizedProperties
-import org.kopi.galite.ui.vaadin.base.VInputButton
+import com.vaadin.flow.component.icon.VaadinIcon
 
 /**
  * Confirm type notification component.
+ *
+ * @param title the confirm notification title.
+ * @param message the confirm notification message.
  */
-open class VConfirmNotification(title: String, message: String) : VAbstractNotification() {
+class VConfirmNotification(title: String,
+                           message: String,
+                           locale: String)
+  : VAbstractNotification(title, message, locale) {
 
   //-------------------------------------------------
   // IMPLEMENTATION
   //-------------------------------------------------
 
-  override fun setButtons(locale: String) {
-    ok = VInputButton(LocalizedProperties.getString(locale, "OK"))
-    cancel = VInputButton(LocalizedProperties.getString(locale, "NO"))
-    cancel.addClickListener { hide() }
-    ok.addClickListener { open() }
-    ok.width = "20%"
-    ok.height = "50%"
-    cancel.width = "20%"
-    cancel.height = "50%"
-  }
+  override fun setButtons() {
+    ok = Button(LocalizedProperties.getString(locale, "OK"))
+    cancel = Button(LocalizedProperties.getString(locale, "NO"))
+    cancel.addClickListener { close() }
+    ok.addClickListener { close() }
+    buttons.add(ok)
+    buttons.add(cancel)
 
-  /**
-   * Creates the confirmation notification footer.
-   */
-  override fun createFooter() {
-    footer.add(ok)
-    footer.add(cancel)
-    footer.isSpacing = true
-    footer.justifyContentMode = FlexComponent.JustifyContentMode.CENTER
-    super.setFooter(footer)
+    if (yesIsDefault) {
+      ok.isAutofocus = true
+    } else {
+      cancel.isAutofocus = true
+    }
   }
 
-  override fun onEnterEvent(keyDownEvent: ShortcutEvent?) {
-    cancel.click()
-    cancelFocused = true
-    okFocused = false
-  }
+  override val iconName: VaadinIcon
+    get() = VaadinIcon.QUESTION_CIRCLE
 
-  override fun onRightEvent(keyDownEvent: ShortcutEvent?) {
+  fun onArrowRightEvent(keyDownEvent: ShortcutEvent?) {
     cancel.focus()
-    okFocused = false
-    cancelFocused = true
-
   }
 
-  override fun onLeftEvent(keyDownEvent: ShortcutEvent?) {
+  fun onArrowLeftEvent(keyDownEvent: ShortcutEvent?) {
     ok.focus()
-    okFocused = true
-    cancelFocused = false
   }
 
-
   //------------------------------------------------
   // DATA MEMBERS
   //------------------------------------------------
-  override val iconName: String?
-    get() = "question-circle"
-  var ok = VInputButton()
-  private var cancel = VInputButton()
+  private lateinit var ok: Button
+  private lateinit var cancel: Button
```

Comment:
can we initialize those button here instead of doing that in setButtons function?